src/main/java/db/migration/V166__Ensure_Activities_Of_Numeric_SectionGroups_Are_Associated_To_SectionGroup.java

```diff
@@ -0,0 +1,114 @@
+package db.migration;
+
+import edu.ucdavis.dss.ipa.api.helpers.Utilities;
+import org.flywaydb.core.api.migration.jdbc.JdbcMigration;
+
+import java.sql.*;
+
+public class V166__Ensure_Activities_Of_Numeric_SectionGroups_Are_Associated_To_SectionGroup implements JdbcMigration {
+
+    @Override
+    public void migrate(Connection connection) throws Exception {
+        // Find all activities
+        // Is the activity associated to a section?
+        // If the section has a numeric sequenceNumber, add a sectionGroup association and remove the section association
+
+        PreparedStatement psActivities = connection.prepareStatement("SELECT * FROM `Activities`");
+
+        try {
+            connection.setAutoCommit(false);
+
+            ResultSet rsActivities = psActivities.executeQuery();
+
+            while (rsActivities.next()) {
+                long activityId = rsActivities.getLong("Id");
+                long sectionId = rsActivities.getLong("SectionId");
+                long sectionGroupId = rsActivities.getLong("SectionGroupId");
+
+                if (sectionGroupId > 0) {
+                    continue;
+                }
+
+                // Find Section
+                PreparedStatement psSection = connection.prepareStatement(
+                        "SELECT * FROM `Sections` WHERE `Id` = ? ;"
+                );
+
+                psSection.setLong(1, sectionId);
+
+                ResultSet rsSection = psSection.executeQuery();
+                while (rsSection.next()) {
+                    sectionGroupId = rsSection.getLong("SectionGroupId");
+                }
+
+                // Find SectionGroup
+                long courseId = 0L;
+                String sequencePattern = null;
+
+                PreparedStatement psSectionGroup = connection.prepareStatement(
+                    "SELECT * FROM `SectionGroups` WHERE `Id` = ? ;"
+                );
+
+                psSectionGroup.setLong(1, sectionGroupId);
+
+                ResultSet rsSectionGroup = psSectionGroup.executeQuery();
+                while (rsSectionGroup.next()) {
```

Comment:
You don't need a while() loop since there should only be one result with that Id (dup).